### DISCUSSION TITLE
Convert / move to expenses

### DISCUSSION THREAD

**Discussion by praul** [kimai/kimai#3132] [Questions & Answers]
**Is your feature request related to a problem? Please describe.**
I recently bought the expenses plugin. I still have a lot of timesheet entries (dated in the future) with fixed amounts. I use these to bill hosting to my clients. Since I now heard about the upcoming deprecation of fixed amounts, I would like to convert them to expenses, as easy as possible.

**Describe the solution you'd like**
Convert all entries of one activity/user to expenses
Alternatively, one-click convert of single entries.

**Comment by kevinpapst** [comment]
Thanks for buying the expense plugin and supporting Kimai!

Ok, there are several possible solutions:
- add a "migrate" button in the UI (a lot of work for a questionable amount of users and a one-time process)
- provide some simple SQL statements to transfer the entries

Let start with the seconds option by querying all timesheets for the ones with a fixed rate:
```sql
select * from kimai2_timesheet where fixed_rate > 0;
```
Or only the ones from the future:
```sql
select * from kimai2_timesheet where fixed_rate > 0 and start_time > now();
```
This helps identifying the timesheets which should be transferred to expenses. 

The real transfer comes later, I will update this answer and leave a comment soon...

**Reply by praul** [reply]
This would be fine by me. Could this be a console command or maybe a php file I can run inside the kimai container (I use docker)?
I'm not used to handle mysql in a console :)
